packages/components/src/components/common/toggleGroup/toggleGroup.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import * as ToggleGroupPrimitive from '@radix-ui/react-toggle-group';
+
+import { cn } from '@/utils/class-name';
+import { toggleBar } from './toggleGroup.css';
+import { useReadonly } from '@/context';
+import { useField } from '@/components';
+
+type ToggleBarProps = React.ComponentPropsWithoutRef<typeof ToggleGroupPrimitive.Root> & {
+  gap?: 1 | 2 | 3 | 4 | undefined;
+};
+
+const ToggleGroup = React.forwardRef<React.ElementRef<typeof ToggleGroupPrimitive.Root>, ToggleBarProps>(
+  ({ disabled, className, gap, ...props }, ref) => {
+    const readonly = useReadonly();
```

Comment:
not sure if this component need to listen to the global readonly context...
Does this need to be disabled, if e.g. the form editor is readonly? I think not, so I would remove it

Reply:
yes it can. just set the disabled flag ;)